language-plutus-core/create-cost-model/CreateModel.hs

```diff
@@ -0,0 +1,143 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE QuasiQuotes         #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Main where
+
+import           Foreign.R
+-- import qualified Foreign.R.Type as R
+import           Control.Exception                                  (TypeError (..))
+import           Control.Monad.Catch
+import           Data.Aeson
+import           Data.Aeson.Encode.Pretty
+import qualified Data.ByteString.Lazy                               as BSL
+import           Data.Default
+import           Data.Either.Extra
+import           Data.Text                                          as T
+import qualified Data.Text.Encoding                                 as T
+import           Data.Vector
+import           H.Prelude                                          (MonadR, Region, r)
+import           Language.PlutusCore.Evaluation.Machine.ExBudgeting
+import           Language.R
+
+main :: IO ()
+main = do
+  model <- withEmbeddedR defaultConfig $ runRegion $ do
+      -- TODO git rev-parse --show-toplevel
+      benchData <- [r|
+        library(jsonlite)
+        source("budgeting-bench/benchData.R")
+        benchData("budgeting-bench/csvs/benching.csv")
+        |]
+      paramAddInteger <- addInteger benchData
+      paramSubtractInteger <- subtractInteger benchData
+      paramMultiplyInteger <- multiplyInteger benchData
+      paramDivideInteger <- divideInteger benchData
+      paramQuotientInteger <- quotientInteger benchData
+      paramRemainderInteger <- remainderInteger benchData
+      paramModInteger <- modInteger benchData
+      paramLessThanInteger <- lessThanInteger benchData
+      paramLessThanEqInteger <- lessThanEqInteger benchData
+      paramGreaterThanInteger <- greaterThanInteger benchData
+      paramGreaterThanEqInteger <- greaterThanEqInteger benchData
+      paramEqInteger <- eqInteger benchData
+      paramConcatenate <- concatenate benchData
+      paramTakeByteString <- takeByteString benchData
+      paramDropByteString <- dropByteString benchData
+      paramSHA2 <- sHA2 benchData
+      paramSHA3 <- sHA3 benchData
+      paramVerifySignature <- verifySignature benchData
+      paramEqByteString <- eqByteString benchData
+      paramLtByteString <- ltByteString benchData
+      paramGtByteString <- gtByteString benchData
+      paramIfThenElse <- ifThenElse benchData
```

Comment:
This is kind of repetitive.  Could we use TH or something to simplify this, or would that be overkill?  (I don't have any particular opinion here, just wondering if that sort of thing's reasonable.)

Reply:
If I had `sequenceRecord` from PureScript, I'd whip something up easily, but here I'm not sure if the benefits would make it worth it. Smaller code is cool when you expect heavy modification, but I see this one to be write-once mostly. So I'd vote for this version until I'll have to mess with it again, then I'll optimize it.